### DISCUSSION TITLE
Ways to Detect Duple-Checkmate but not Allow Directly Capturing a Pseudo-Royal Piece

### DISCUSSION THREAD

**Discussion by PraseodymiumSpike** [fairy-stockfish/Fairy-Stockfish#819] [Q&A]
Is there any way to detect duple-checkmate but not allow directly capturing a pseudo-royal piece? I wanted to make a game similar to Spartan Chess, but where the chess side has to win by duple-checkmate and can't just capture 1 royal piece and checkmate the other. To detect duple-checkmate, extinction is required, but the only way to not allow capturing just 1 royal piece is to make the opponent lose when they capture it, which also requires extinction! Furthermore, these 2 uses conflict! I tried to just make it so the number of pieces required for extinction to be declared is 1 meaning that when 1 pseudo-royal piece is taken, the taker will lose, but that means check triggers when either pseudo-royal piece is attacked because the second-to-last royal piece is considered the last one before extinction is declared. Now, I tried other ways to detect when a piece count drops below 2, but none worked. Connection lets you detect when a piece is on the board, but it doesn't let you detect when a piece is captured. The same is true of capture-the-flag rules. I even tried making pieces go to hand and forcing the pseudo-royal piece to be dropped whenever possible, then making it impossible to drop them anywhere, hoping to cause a stalemate, and then, I made stalemate a loss. But it doesn't force you to drop when there is nowhere to drop, so that didn't work either! Is there any way to make this work?

**Comment by ianfab** [comment]
Not sure whether this is possible at the moment. Technically if you do the same as you did but set the extinction count to 0 (or maybe even -1?) it should still detect duple check but not consider pseudo royals yet, so maybe that works a bit better.

**Reply by PraseodymiumSpike** [reply]
I think it would allow the chess side to capture 1 of the pseudo-royal pieces without immediately losing, then, wouldn't it? The `extinctionValue = win` wouldn't trigger when 1 is captured.

**Reply by ianfab** [reply]
Yes, but it would still be disincentiviced to take strategically. I guess that is the lesser of the evils compared to the other options.

**Reply by PraseodymiumSpike** [reply]
Well, I have made an issue for it for now.

**Comment by PraseodymiumSpike** [comment]
okay i kind of figured out a hacky way to do this but it allows taking one pseudoroyal if this simultaneously mates the other
```
[yadt:crazyhouse]
king = -
alfil = a
commoner = k
maxFile = 11
extinctionPieceTypes = n
extinctionPseudoRoyal = true
dupleCheck = true
mustDrop = true
mustDropType = n
connectRegion1White = k1
connectRegion2White = k2
connectRegion1Black = k3
connectRegion2Black = k4
connectValue = loss
connectPieceTypes = pnbrqka
startFen = rnbqkbnr***/pppppppp***/8***/8***/8**1/8**a/PPPPPPPP**1/RNBQKBNR**A[] w KQkq - 0 1
whiteDropRegion = k2
blackDropRegion = k4
castlingRookKingsideFile = h
castlingKingPiece = k
```
this works by forcing you to lose by connection by forcing you to drop a royal (in this case a knight) after you take one in the one place that makes you lose by connection (which is the only place you are allowed to drop)

ofc another piece can be dropped there too but only knights are required, and since dropping makes you lose instantly, no one would ever drop there

i wish it didnt allow taking one pseudoroyal while mating the other but ill take what i can get

**Reply by PraseodymiumSpike** [reply]
btw yadt isnt a serious variant, its just a proof of concept for this method
